### PR TITLE
Added functionality to test_suite_daemon.py about tensorboard plots, …  … …customized regularization, learning rate and epochs-to-train-before-validating-once, while still maintaining backward compatibility with test_suite.py

### PR DESCRIPTION
**Pull request by bohanwu**

### CODE REVIEW

File: models/inception_resnet.py [hunk 1]
```diff
@@ -270,7 +270,7 @@ def inception_res_model(input, num_A, num_B, num_C, num_classes, is_training):
 
   return output
 
-def setup_resnet_inception_model(image_size, num_classes, A, B, C, learning_rate=1e-3):
+def setup_resnet_inception_model(image_size, num_classes, A, B, C, learning_rate=1e-3, reg=0):
```
Comment (tmaudev) [inline models/inception_resnet.py]: What's the point of this reg term? We don't use it

File: utilities/train.py [hunk 2]
```diff
@@ -1,25 +1,28 @@
 import math
-import tensorflow as tf
-import numpy as np
 import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
```
Comment (tmaudev) [inline utilities/train.py]: Just curious, what is the point of this? lol

File: utilities/train.py [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
+
 
 # Save Checkpoint of Model
 def save_model_checkpoint(session, saver, filename, epoch_num):
   save_path = saver.save(session, filename, epoch_num)
   print("\nCheckpoint saved in file: %s" % save_path)
 
+
```
Comment (tmaudev) [inline utilities/train.py]: What is the point having all these unnecessary newlines?

File: tests/test_inception_resnet_2d.py [hunk 4]
```diff
@@ -17,10 +20,44 @@ def run_inception_resnet_2d_test(data, num_classes, device, recover, ckpt_path,
...
+    global_step += train_epochs_per_validation - 1
+
+    # Validate Model
+    print("Validating model...")
+    train_model(device, sess, model, data['X_val'], data['y_val'], epochs=1, batch_size=64, is_training=False,
+                log_freq=100, plot_loss=False, global_step=global_step, writer=val_writer)
+
+    global_step += 1
+
+  print("----------TRAINING IS DONE!-----------")
```
Comment (tmaudev) [inline tests/test_inception_resnet_2d.py]: Do we need this?

File: utilities/train.py (hunk 3 above)
Comment (bohanwu) [inline utilities/train.py]: it's not necessary

File: tests/test_inception_resnet_2d.py (hunk 4 above)
Comment (bohanwu) [inline tests/test_inception_resnet_2d.py]: the global step Yes, the prints? deleting is okay.

File: models/inception_resnet.py (hunk 1 above)
Comment (bohanwu) [inline models/inception_resnet.py]: Some kine of regularization will be needed for later models, but for now I will not add additional arguments into InceptionResnet, but I will still add an argument option of dropout into Test_suite_daemon

File: utilities/train.py (hunk 2 above)
Comment (bohanwu) [inline utilities/train.py]: Let me check if this is used in the file

File: tests/test_inception_resnet_2d.py (hunk 4 above)
Comment (tmaudev) [inline tests/test_inception_resnet_2d.py]: Yeah I meant the print lol

File: utilities/train.py (hunk 2 above)
Comment (tmaudev) [inline utilities/train.py]: I mean the point of reordering things lol

Comment (bohanwu) [inline utilities/train.py]: then it's not necessary

File: data/uwash_rgbd/load_pickles.py [hunk 5]
```diff
@@ -93,6 +93,7 @@ def package_data(X_train, y_train, X_test_val, y_test_val, depth):
         max_train = np.max(X_train[:, :, :, 3], axis=(0,1,2))
         X_train[:, :, :, 3] *= (255.0 / max_train)
         X_test_val[:, :, :, 3] *= (255.0 / max_train)
+        max_train = None
```
Comment (iKuba) [inline data/uwash_rgbd/load_pickles.py]: why bother?

File: tests/test_inception_resnet_2d.py [hunk 6]
```diff
@@ -17,10 +20,44 @@ def run_inception_resnet_2d_test(data, num_classes, device, recover, ckpt_path,
...
+  global_step = 0
+
+  for i in range(num_train_val_cycles):
+    # Train Model
+    print("Training model...")
+    train_model(device, sess, model, data['X_train'], data['y_train'], epochs=train_epochs_per_validation,
+                batch_size=64, is_training=True, log_freq=100, plot_loss=False, global_step=global_step,
+                writer=train_writer)
+
+    global_step += train_epochs_per_validation - 1
```
Comment (iKuba) [inline tests/test_inception_resnet_2d.py]: this seems mad awkward to decrement then increment

File: tests/test_inception_resnet_2d.py [hunk 7]
```diff
@@ -17,10 +20,44 @@ def run_inception_resnet_2d_test(data, num_classes, device, recover, ckpt_path,
...
+
+    # Validate Model
+    print("Validating model...")
+    train_model(device, sess, model, data['X_val'], data['y_val'], epochs=1, batch_size=64, is_training=False,
+                log_freq=100, plot_loss=False, global_step=global_step, writer=val_writer)
```
Comment (iKuba) [inline tests/test_inception_resnet_2d.py]: I'd suggest you just do global_step = global_step - 1 here, it will be more readable and make it less mysterious why you decrement global_step by one then increment.

But I'm just being nitpicky.

File: tests/test_suite_daemon.py [hunk 8]
```diff
@@ -14,115 +14,125 @@
...
+  debug = args.debug
+
+  print("\nLoading data...")
+  # Get Appropriate Data
+  if dataset == 'cifar':
+    # Get CIFAR-10 Dataset
+    data = get_CIFAR10_data(num_training=49000, num_validation=1000,
+                            num_test=1000,
+                            subtract_mean=True)
+    num_classes = 10
```
Comment (iKuba) [inline tests/test_suite_daemon.py]: I think this could be inferred but hardcoding is fine I guess.

File: utilities/train.py (hunk 2 above)
Comment (bohanwu) [inline utilities/train.py]: the order is auto-reformatted, because n is before t.

File: data/uwash_rgbd/load_pickles.py (hunk 5 above)
Comment (bohanwu) [inline data/uwash_rgbd/load_pickles.py]: this is so save memory - you guys have monster machines of 128G of RAM and I don't. Plus, imagine if you are running 8 processes individually on 8 GPUs on experiment simultaneously. If we don't have these None statements the programs will crash.

File: tests/test_suite_daemon.py (hunk 8 above)
Comment (bohanwu) [inline tests/test_suite_daemon.py]: yup this was not my edit

**Comment by bohanwu** [comment]
@iKuba @tmaudev feel free to review this MR again

File: tests/test_suite_daemon.py (hunk 8 above)
Comment (tmaudev) [inline tests/test_suite_daemon.py]: YOLO

File: tests/test_inception_resnet_2d.py (hunk 6 above)
Comment (bohanwu) [inline tests/test_inception_resnet_2d.py]: It's okay
